ksql-common/src/test/java/io/confluent/ksql/schema/connect/SchemaWalkerTest.java

```diff
@@ -0,0 +1,330 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.schema.connect;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.clearInvocations;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.common.collect.Streams;
+import io.confluent.ksql.schema.connect.SchemaWalker.Visitor;
+import java.util.stream.Stream;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Spy;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SchemaWalkerTest {
+
+  @Spy
```

Comment:
I put this comment on the main PR, but I'm not sure I understand your desire to steer clear of abstract classes - this seems like a perfect candidate (and follows the reset of our visitor pattern classes)

Reply:
Abstract classes have disadvantages over interfaces, e.g. no multiple inheritance.

I favour composition over inheritance 99% of the time as it has much less gotchas.